<launch>
	
	<!--Others-->

	<param name="use_sim_time" value="false" />

	<!--node pkg="laser_ekf" type="odom_zh" name="odom_zh"/-->
	<node pkg="laser_ekf" type="odom_fake" name="odom_fake"/>

	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find laser_ekf)/config/sliding_map_dynamic.rviz" output="screen"/>	

	


	<!--New sensor set, New calibration needed.-->

	<!--node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="1.95685 -2.19474 -0.1 -0.017 -0.0104 -0.521 -0.853 laser2 laser1 50" />
    <node pkg="tf" type="static_transform_publisher" name="link2_broadcaster" args="7.5012 1.4813 0.962 0 0 -0.3625 0.9320 base_footprint laser2 50" /-->

    <node pkg="tf" type="static_transform_publisher" name="link2_broadcaster" args="7.5012 -1.4813 -0.962 0 0 -0.3625 0.9320 base_footprint laser2 50" />
	<node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="-2.05685 1.99474 0 0.0000 0.0000 0.7880 0.6157 laser2 laser1 100" />

	



	<param name="robot_description" textfile="$(find laser_ekf)/config/zhenhua_vehicle.urdf" />
	<param name="use_gui" value="false"/>
	<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
	<node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />

        <arg name="publishMapTF" default="false"/>


 


		<node pkg="laser_ekf" type="sliding_mapper_double" name="sliding_mapper_double" output="screen">
		<param name="publishMapTF" value="$(arg publishMapTF)"/>
		<param name="subscribe_scan" value="false"/>
		<param name="subscribe_cloud" value="true"/>
        <param name="subscribe_odom" value="false"/>

		<remap from="scan" to="/scan"/>
		<remap from="cloud_in" to="/velodyne2/velodyne_points"/>
        <remap from="odom_in" to="/wheel_odom"/>
       

		<param name="icpConfig" value="$(find laser_ekf)/config/icp_dynamic.yaml" />
		<param name="inputFiltersConfig" value="$(find laser_ekf)/config/input_filter.yaml" />
		<param name="mapPostFiltersConfig" value="$(find laser_ekf)/config/map_post_filter.yaml" />

        <param name="map_frame" value="map" />    
        <param name="plane_odom_frame" value="odom" />              
		<param name="base_footprint_frame" value="base_footprint"/>   

		<param name="useROSLogger" value="false" />
		<param name="inputQueueSize" value="1" />
		<param name="tfRefreshPeriod" value="0.005" />
		<param name="minOverlap" value="0.2" /> 
		<param name="maxOverlapToMerge" value="0.99" /> 
		<param name="minReadingPointCount" value="500" /> 
		<param name="minMapPointCount" value="500" />    <!--10000-->
		<param name="localizing" value="true" /> 
		<param name="mapping" value="true" />

        <param name="sensorMaxRange" value="80"/>  <!--80-->
        <param name="slidingMapMaxRange" value="30"/>  <!--85-->

		<param name="priorStatic" value="0.45"/>  <!--0.45-->
		<param name="priorDyn" value="0.55"/>     <!--0.55-->
		<param name="maxAngle" value="0.01"/>  <!--0.01 :(rad)  Angle around which all points in Q are associated to a point p-->
		<param name="eps_a" value="0.1"/>      <!--0.1  :      ratio of measurement noise propotional with range -->  
		<param name="eps_d" value="0.1"/>      <!--0.1  :(0.1m) fixed noisy part of measurement-->
		<param name="alpha" value="0.99"/>     <!--0.99 : Probability of staying static given that the point was static-->
		<param name="beta" value="0.90"/>      <!--0.90 : Probability of staying dynamic given that the point was dynamic-->
		<param name="maxDyn" value="0.90"/>    <!--0.90 : Probability at which a point is considered permanently dynamic-->
        <param name="dynThreshold" value="0.85"/>

		<param name="maxDistNewPoint" value="0.5"/>  <!--0.7 -->
	</node>	

</launch>
